lib/components/Forms/Fields/NumberInput/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { forwardRef, useEffect, useState } from "react"
+import { Input, InputProps } from "../Input"
+
+const FORMAT = /^([0-9]+)$/
+// eslint-disable-next-line no-useless-escape
+const FORMAT_WITH_DECIMALS = /^([0-9]+)(?:[\.,]([0-9]+)?)?$/
+
+const parseValue = (value: string) => parseFloat(value.replace(",", "."))
+
+type NumberInputProps = Omit<InputProps, "value" | "type" | "onChange"> & {
+  locale: string
```

Comment:
The locale is a required prop, I'm wondering if there's a better approach...

Reply:
we can have locale undefined. according to the docs it will use the locale of the runtime (i.e. browser) will be used:
> The runtime's default locale is used when undefined is passed or when none of the specified locale identifiers is supported.

Reply:
Yes, but making the locale prop optional can lead us to inconsistencies when the browser is in a different language than the user-defined language in Factorial, people will eventually forget to pass the prop if it's not required...

Reply:
I agree with vinicius, I wouldn't make it optional. If this component was in factorial we could do it because we can access to the employee locale, but here...

Reply:
Thanks for this learning, folks!